src/NuGet.Core/NuGet.Protocol/Model/GetVulnerabilityInfoResult.cs

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+
+namespace NuGet.Protocol.Model
+{
+    /// <summary>Represents the result of <see cref="IVulnerabilityInfoResource.GetVulnerabilityInfoAsync(Core.Types.SourceCacheContext, Common.ILogger, System.Threading.CancellationToken)"/></summary>
+    /// <remarks>Vulnerability data from a package source typically is spread across multiple files. Therefore, it's
+    /// possible for some data to be retrieved successfully, while other data failed. This type allows partial data
+    /// to be returned along with error information.</remarks>
+    public class GetVulnerabilityInfoResult
```

Comment:
From https://github.com/NuGet/NuGet.Client/pull/5115#issuecomment-1491164057, since GitHub mysteriously doesn't supported threaded/resolvable conversations unless it's linked to a specific line in a file

> I am curious to understand why not return VulnerabilityInfoResult instead of GetVulnerabilityInfoResult

@kartheekp-ms my thought process was that the data type for the payload data (list of dictionary) is very specific to one API and might not be interesting to other APIs. So, rather than using an overly generic name for a specific data type, give a specific name.  Although there's a good chance I was over-thinking it.

I also experimented with:

``` cs
public class ApiResult<T>
{
    public T? Result { get; }
    public IReadOnlyList<Exception>? Exceptions { get; }
}
```

However, I had some problem trying to get it to work. I no longer remember exactly what problems, possibly something to do with Newtonsoft.Json deserialization.